app/controllers/session_controller.rb

```diff
@@ -0,0 +1,18 @@
+class SessionController < ApplicationController
+  def create
+    user = User.from_omniauth(env["omniauth.auth"])
+    session[:user_id] = user.id
+    flash[:notice] = "Logged In !!"
+    redirect_to root_path
+  end
+
+  def auth_hash
+  	request.env['omniauth.auth']
```

Comment:
Use 2 (not 1) spaces for indentation.<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.<br>Tab detected.